app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+
+  validates_presence_of :username
+  validates_uniqueness_of :username
```

Comment:
Rails/Validation: Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.